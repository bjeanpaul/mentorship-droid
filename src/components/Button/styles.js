import { StyleSheet } from 'react-native';
import colors from 'src/constants/colors';
import { FONT, FONT_WEIGHT } from 'src/constants/styles';

export default StyleSheet.create({
  container: {
    justifyContent: 'center',
    alignItems: 'center',
    borderRadius: 5,
  },

  title: {
    fontFamily: FONT.BOLD,
    fontWeight: FONT_WEIGHT.BOLD,
    fontSize: 13,
    letterSpacing: 1,
    marginTop: 15,
    marginBottom: 17,
  },
});


export const layouts = StyleSheet.create({
  default: {
    marginLeft: 16,
    marginRight: 16,
  },

  inline: {
    marginLeft: 0,
    marginRight: 0,
  },

  stretch: {
    alignSelf: 'stretch',
  },
});


export const themes = {
  default: StyleSheet.create({
    container: {
      backgroundColor: colors.BUTTON_BG,
    },

    title: {
      color: colors.BUTTON_TITLE,
    },

    containerIsDisabled: {
      backgroundColor: colors.BUTTON_DISABLED_BG,
    },

    titleIsDisabled: {
      color: colors.BUTTON_DISABLED_TITLE,
    },
  }),


  light: StyleSheet.create({
    container: {
      backgroundColor: colors.BUTTON_LIGHT_BG,
    },

    title: {
      color: colors.BUTTON_LIGHT_TITLE,
    },

    containerIsDisabled: {
      backgroundColor: colors.BUTTON_DISABLED_BG,
    },

    titleIsDisabled: {
      color: colors.BUTTON_DISABLED_TITLE,
    },
  }),


  transparent: StyleSheet.create({
    container: {
      backgroundColor: colors.BUTTON_TRANSPARENT_BG,
      borderColor: colors.BUTTON_TRANSPARENT_BORDER,
      borderWidth: 1,
    },

    title: {
      color: colors.BUTTON_TRANSPARENT_TITLE,
    },

    containerIsDisabled: {
      borderColor: colors.BUTTON_TRANSPARENT_DISABLED_BORDER,
    },

    titleIsDisabled: {
      color: colors.BUTTON_TRANSPARENT_DISABLED_TITLE,
    },
  }),
};
