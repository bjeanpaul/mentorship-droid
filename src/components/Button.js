import React, { PropTypes } from 'react';
import { TouchableNativeFeedback, View, StyleSheet } from 'react-native';

import Text from 'Text';
import { font, fontWeight } from 'src/app/styles';


const styles = StyleSheet.create({
  button: {
    justifyContent: 'center',
    alignItems: 'center',
    backgroundColor: '#f77040',
    borderRadius: 5,
    marginLeft: 16,
    marginRight: 16,
  },
  buttonDisabled: {
    backgroundColor: '#dfe5e6',
  },
  label: {
    color: '#fff',
    fontFamily: font.brandonTextBold,
    fontSize: 13,
    fontWeight: fontWeight.bold,
    letterSpacing: 1,
    marginTop: 15,
    marginBottom: 17,
  },
});

const Button = ({
  label,
  disabled,
  handlePress,
}) => {
  let button = (
    <View style={[styles.button, disabled && styles.buttonDisabled]}>
      <Text style={styles.label}>{label.toUpperCase()}</Text>
    </View>
  );
  if (!disabled) {
    button = (
      <TouchableNativeFeedback onPress={handlePress}>
        {button}
      </TouchableNativeFeedback>
    );
  }
  return button;
};
Button.propTypes = {
  label: PropTypes.string.isRequired,
  handlePress: PropTypes.func.isRequired,
  disabled: PropTypes.bool,
};
export default Button;
