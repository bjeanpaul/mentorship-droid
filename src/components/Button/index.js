import React, { PropTypes } from 'react';
import { TouchableNativeFeedback, View } from 'react-native';
import Text from 'src/components/Text';
import styles, { themes, layouts } from './styles';


const Button = ({
  theme = themes.default,
  layout = layouts.default,
  children,
  disabled,
  onPress,
}) => {
  let button = (
    <View
      style={[
        styles.container,
        theme.container,
        layout,
        disabled && theme.containerIsDisabled,
      ]}
    >
      <Text
        style={[
          styles.title,
          theme.title,
          disabled && theme.titleIsDisabled,
        ]}
      >
        {children}
      </Text>
    </View>
  );
  if (!disabled) {
    button = (
      <TouchableNativeFeedback onPress={onPress}>
        {button}
      </TouchableNativeFeedback>
    );
  }
  return button;
};


Button.themes = themes;
Button.layouts = layouts;


Button.propTypes = {
  children: PropTypes.any,
  onPress: PropTypes.func.isRequired,
  disabled: PropTypes.bool,
  theme: PropTypes.any,
  layout: PropTypes.any,
};


export default Button;
