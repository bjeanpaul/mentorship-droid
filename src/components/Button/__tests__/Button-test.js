import React from 'react';
import Button from 'src/components/Button';


describe('Button', () => {
  function foo() {}

  it('should render its `title,` and map `onPress`', () => {
    expect(render(
      <Button
        label="margle"
        onPress={foo}
      />
    )).toMatchSnapshot();
  });

  it('should render "light" theme', () => {
    expect(render(
      <Button
        label="margle"
        onPress={foo}
        theme={Button.themes.light}
      />
    )).toMatchSnapshot();
  });

  it('should render "transparent" theme', () => {
    expect(render(
      <Button
        label="margle"
        onPress={foo}
        theme={Button.themes.transparent}
      />
    )).toMatchSnapshot();
  });

  it('should render "inline" layout', () => {
    expect(render(
      <Button
        label="margle"
        onPress={foo}
        layout={Button.layouts.inline}
      />
    )).toMatchSnapshot();
  });

  it('should be disabled', () => {
    expect(render(
      <Button
        label="margle"
        onPress={foo}
        disabled
      />
    )).toMatchSnapshot();
  });

  it('should be able to tap and fire `onPress`', () => {
    const mockFn = jest.fn();
    const el = shallow(
      <Button
        label="margle"
        onPress={mockFn}
      />
    );
    el.simulate('press');
    expect(mockFn).toBeCalled();
  });
});
