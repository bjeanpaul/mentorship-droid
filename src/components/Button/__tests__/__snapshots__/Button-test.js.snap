exports[`Button should be disabled 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "borderRadius": 5,
        "justifyContent": "center",
        "marginHorizontal": 16,
        "paddingBottom": 17,
        "paddingTop": 15,
      },
      Object {
        "backgroundColor": "#f77040",
      },
      undefined,
      undefined,
      Array [
        Object {
          "backgroundColor": "#dfe5e6",
        },
        undefined,
      ],
    ]
  }>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Array [
        Object {
          "color": "#003035",
          "fontFamily": "Brandon Text",
        },
        Object {
          "fontSize": 13,
          "textAlign": "center",
        },
        Array [
          Object {
            "fontFamily": "Brandon Text_bold",
            "fontSize": 13,
            "fontWeight": "bold",
            "letterSpacing": 1,
          },
          Object {
            "color": "#ffffff",
          },
          undefined,
          Array [
            Object {
              "color": "#ffffff",
            },
            undefined,
          ],
        ],
      ]
    }>
    margle
  </Text>
</View>
`;

exports[`Button should render "inline" layout 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "backgroundColor": "#ffbcbc",
        "borderColor": "red",
        "borderWidth": 1,
        "height": 100,
        "justifyContent": "center",
        "margin": 10,
        "width": 300,
      },
      undefined,
    ]
  }>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "color": "#333333",
        "margin": 20,
      }
    }>
    TouchableNativeFeedback is not supported on this platform!
  </Text>
</View>
`;

exports[`Button should render "light" theme 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "backgroundColor": "#ffbcbc",
        "borderColor": "red",
        "borderWidth": 1,
        "height": 100,
        "justifyContent": "center",
        "margin": 10,
        "width": 300,
      },
      undefined,
    ]
  }>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "color": "#333333",
        "margin": 20,
      }
    }>
    TouchableNativeFeedback is not supported on this platform!
  </Text>
</View>
`;

exports[`Button should render "small" size 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "backgroundColor": "#ffbcbc",
        "borderColor": "red",
        "borderWidth": 1,
        "height": 100,
        "justifyContent": "center",
        "margin": 10,
        "width": 300,
      },
      undefined,
    ]
  }>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "color": "#333333",
        "margin": 20,
      }
    }>
    TouchableNativeFeedback is not supported on this platform!
  </Text>
</View>
`;

exports[`Button should render "stretch" layout 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "backgroundColor": "#ffbcbc",
        "borderColor": "red",
        "borderWidth": 1,
        "height": 100,
        "justifyContent": "center",
        "margin": 10,
        "width": 300,
      },
      undefined,
    ]
  }>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "color": "#333333",
        "margin": 20,
      }
    }>
    TouchableNativeFeedback is not supported on this platform!
  </Text>
</View>
`;

exports[`Button should render "transparent" theme 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "backgroundColor": "#ffbcbc",
        "borderColor": "red",
        "borderWidth": 1,
        "height": 100,
        "justifyContent": "center",
        "margin": 10,
        "width": 300,
      },
      undefined,
    ]
  }>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "color": "#333333",
        "margin": 20,
      }
    }>
    TouchableNativeFeedback is not supported on this platform!
  </Text>
</View>
`;

exports[`Button should render its \`title,\` and map \`onPress\` 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "backgroundColor": "#ffbcbc",
        "borderColor": "red",
        "borderWidth": 1,
        "height": 100,
        "justifyContent": "center",
        "margin": 10,
        "width": 300,
      },
      undefined,
    ]
  }>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "color": "#333333",
        "margin": 20,
      }
    }>
    TouchableNativeFeedback is not supported on this platform!
  </Text>
</View>
`;

exports[`Button should take any kind of children 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "backgroundColor": "#ffbcbc",
        "borderColor": "red",
        "borderWidth": 1,
        "height": 100,
        "justifyContent": "center",
        "margin": 10,
        "width": 300,
      },
      undefined,
    ]
  }>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "color": "#333333",
        "margin": 20,
      }
    }>
    TouchableNativeFeedback is not supported on this platform!
  </Text>
</View>
`;
